File: testing/cppsrc/leakyrelu.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class leakyrelu {
  public:
    double _alpha;
 
    leakyrelu(double alpha = 0.01) {
      _alpha = alpha;
    }

    vector <double> forward(vector <double> input) {
      transform(input.begin(), input.end(), input.begin(), [&](double input){ return (_alpha * input < input) ? input : 0; });
      return input;
    }

    vector <double> backward(vector <double> input) {
      transform(input.begin(), input.end(), input.begin(), [&](double input){ return (input <= 0) ? _alpha : 1; });
      return input;
    }
};

File: example/src/leakyrelu.py
import numpy as np

class leakyrelu:
  def __init__(self, alpha=0.01):
    self.alpha = alpha

  def forward(self, input):
    return np.maximum(self.alpha * input, input)

  def backward(self, input):
    equation = np.vectorize(self.equationforderivative, otypes=[float])
    return equation(input)

  def equationforderivative(self, input):
    return self.alpha if input < 0 else 1